### ISSUE TITLE
Failed to communicate with "admin/reset" endpoint

### ISSUE THREAD

**Issue by pavetok** [tomakehurst/wiremock#293]
Version: 1.57
Error:
```
com.github.tomakehurst.wiremock.client.VerificationException: Expected status 200 for http://localhost:8080/__admin/reset but was 500
	at com.github.tomakehurst.wiremock.client.HttpAdminClient.postJsonAssertOkAndReturnBody(HttpAdminClient.java:156)
	at com.github.tomakehurst.wiremock.client.HttpAdminClient.resetMappings(HttpAdminClient.java:86)
	at com.github.tomakehurst.wiremock.client.WireMock.resetMappings(WireMock.java:101)
	at com.github.tomakehurst.wiremock.client.WireMock.reset(WireMock.java:105)
	at com.peterservice.project.common.test.stand.WebServerMock.reset(WebServerMock.java:26)
```

Version 1.56 correctly works.

**Comment by benfonty** [comment]
It seems that the problem occurs because the jar is compiled for java 8 and launched by java 7.

I had the same problem and found that the error 500 is caused by  

```java.lang.NoSuchMethodError: java.util.concurrent.ConcurrentHashMap.keySet().```

I think that the problem si the same as [there](https://gist.github.com/AlainODea/1375759b8720a3f9f094).

**Comment by Waxolunist** [comment]
+1 for fixing

**Comment by mbucc** [comment]
I hit this too.

**Comment by skuroda** [comment]
Same problem as #285?

**Comment by tomakehurst** [comment]
I believe this should be fixed in 1.58 and 2.0.8-beta. Please re-open if this isn't the case.
